pkg/apis/kubernetes/v1/crud_test.go

```diff
@@ -0,0 +1,390 @@
+package v1
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/onsi/gomega/ghttp"
+	"go.anx.io/go-anxcloud/pkg/api"
+	"go.anx.io/go-anxcloud/pkg/api/types"
+	corev1 "go.anx.io/go-anxcloud/pkg/apis/core/v1"
+	"go.anx.io/go-anxcloud/pkg/apis/internal/gs"
+	"go.anx.io/go-anxcloud/pkg/client"
+	"go.anx.io/go-anxcloud/pkg/utils/pointer"
+	testutils "go.anx.io/go-anxcloud/pkg/utils/test"
+)
+
+const (
+	mockKubernetesVersion = "1.23.6"
+	nonExistingIdentifier = "non-existing-identifier"
+)
+
+var _ = Describe("CRUD", Ordered, func() {
+	var (
+		a   api.API
+		srv *ghttp.Server
+
+		clusterIdentifier  string
+		nodePoolIdentifier string
+
+		clusterName  = "go-anxcloud-" + testutils.RandomHostname()
+		nodePoolName = "go-anxcloud-" + testutils.RandomHostname()
+		location     = corev1.Location{Identifier: "52b5f6b2fd3a4a7eaaedf1a7c019e9ea"} // ANX04
+	)
+
+	mockStateOK := map[string]interface{}{"id": "0"}
+	mockStateError := map[string]interface{}{"id": "1"}
+	mockStatePending := map[string]interface{}{"id": "2"}
+
+	BeforeEach(func() {
+		srv = ghttp.NewServer()
+
+		var err error
+
+		if isIntegrationTest {
+			a, err = api.NewAPI(api.WithClientOptions(client.AuthFromEnv(false)))
+			Expect(err).ToNot(HaveOccurred())
+		} else {
+			a, err = api.NewAPI(api.WithClientOptions(
+				client.BaseURL(srv.URL()),
+				client.IgnoreMissingToken(),
+			))
+			Expect(err).ToNot(HaveOccurred())
+		}
+	})
+
+	Context("Cluster object", Ordered, func() {
+		Context("Create operation", Ordered, func() {
+			It("Create new cluster", func() {
+				appendCreateClusterHandler(srv,
+					map[string]interface{}{"name": clusterName, "location": location.Identifier},
+					mockedClusterResponse(Cluster{Identifier: "mocked-cluster-identifier", Name: clusterName}),
+				)
+
+				cluster := Cluster{Name: clusterName, Location: location}
+
+				err := a.Create(context.TODO(), &cluster)
+				Expect(err).ToNot(HaveOccurred())
+				Expect(cluster.Identifier).NotTo(BeEmpty())
+				clusterIdentifier = cluster.Identifier
+				Expect(cluster.StatePending()).To(BeTrue())
+				Expect(pointer.BoolVal(cluster.NeedsServiceVMs)).To(BeTrue()) // Default if not set on Create
+			})
+		})
+
+		Context("AwaitCompletion", Ordered, func() {
```

Comment:
this full block could move into unit tests, not needing the Engine to test at all - all Engine stuff is tested with other specs here (like testing `Get`).